src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -236,6 +301,80 @@ protected static async Task<ImmutableArray<SyntaxToken>> GetIdentifierOrGlobalNa
             };
         }
 
+        protected static async Task<ImmutableArray<FinderLocation>> FindReferencesInDocumentInsideGlobalSuppressionsAsync(
+            Document document,
+            SemanticModel semanticModel,
+            ISyntaxFacts syntaxFacts,
+            Func<string> getDocumentationCommentId,
+            CancellationToken cancellationToken)
+        {
+            var info = await SyntaxTreeIndex.GetIndexAsync(document, cancellationToken).ConfigureAwait(false);
+            if (!info.ContainsGlobalAttributeInfo)
+            {
+                return ImmutableArray<FinderLocation>.Empty;
+            }
+
+            var suppressMessageAttribute = semanticModel.Compilation.SuppressMessageAttributeType();
+            if (suppressMessageAttribute == null)
+            {
+                return ImmutableArray<FinderLocation>.Empty;
+            }
+
+            var documentationCommentId = getDocumentationCommentId();
+            if (string.IsNullOrEmpty(documentationCommentId) ||
+                !info.TryGetPositionsInGlobalAttributes(documentationCommentId, out var positions))
```

Comment:
this call looks super weird.

Reply:
Renamed for clarity and also added doc comments.